torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -97,7 +97,7 @@ static Variable applySelect(const Variable& self, int64_t dim, int64_t index) {
   int64_t size = self.size(dim);
   if (index < -size || index >= size) {
     throw IndexError("index %lld is out of bounds for dimension %lld with size %lld",
-      index, dim, size);
+      index, self.dim(), size);
```

Comment:
I think this is incorrect, `self.dim()` returns the number of dimensions of the tensor IIRC.

Reply:
You are right @vishwakftw 
Instead, full variable needs to be passed from applySlicing and update dimension